### PR TITLE
Alerting UX: update descriptions for alerting notification fields

### PR DESCRIPTION
**Pull request by ppcano**
1 - This PR aligns the descriptions with the updated [Group notifications and Timing options docs](https://grafana.com/docs/grafana/latest/alerting/fundamentals/notifications/group-alert-notifications/). 

2 - I wanted to include `NeedHelp?` that links to [Group alert notifications docs](https://grafana.com/docs/grafana/next/alerting/fundamentals/notifications/group-alert-notifications/) in the `Override grouping` and `Override timing` options, but was unsure how.  But we have recently discussed to change the UI for both options, so it may then be a better moment. 

Screenshot 2024-06-13 at 19 45 11

### CODE REVIEW

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. Just letting you know I'm updating your PR's milestone to [11.2.x](https://github.com/grafana/grafana/milestone/529). If you need to see your changes in a previous version, please consider adding a backport label.

Thank you!

File: public/app/features/alerting/unified/components/notification-policies/EditDefaultPolicyForm.tsx [hunk 1]
```diff
@@ -133,7 +133,7 @@ export const AmRootRouteForm = ({
           </Field>
           <Field
             label="Group interval"
-            description="The waiting time to send a batch of new alerts for that group after the first notification was sent. Default 5 minutes."
+            description="The waiting time before sending a notification about changes in the alert group after the first notification has been sent. Default 5 minutes."
```
Comment (brendamuir) [inline public/app/features/alerting/unified/components/notification-policies/EditDefaultPolicyForm.tsx:136]:
```suggestion
            description="The wait time before sending a notification about changes in the alert group after the first notification has been sent. Default is 5 minutes."
```

File: public/app/features/alerting/unified/components/notification-policies/EditDefaultPolicyForm.tsx [hunk 2]
```diff
@@ -147,7 +147,7 @@ export const AmRootRouteForm = ({
           </Field>
           <Field
             label="Repeat interval"
-            description="The waiting time to resend an alert after they have successfully been sent. Default 4 hours. Should be a multiple of Group interval."
+            description="The waiting time before resending a notification that has already been sent successfully. Default 4 hours. Should be a multiple of Group interval."
```
Comment (brendamuir) [inline public/app/features/alerting/unified/components/notification-policies/EditDefaultPolicyForm.tsx:150]:
```suggestion
            description="The wait time before resending a notification that has already been sent successfully. Default is 4 hours. Should be a multiple of Group interval."
```

File: public/app/features/alerting/unified/components/notification-policies/EditNotificationPolicyForm.tsx [hunk 3]
```diff
@@ -191,7 +191,7 @@ export const AmRoutesExpandedForm = ({
       {watch().overrideGrouping && (
         <Field
           label="Group by"
-          description="Group alerts when you receive a notification based on labels. If empty it will be inherited from the parent policy."
+          description="Combine multiple alerts into a single notification by grouping them by the same label values. If empty it will be inherited from the parent policy."
```
Comment (brendamuir) [inline public/app/features/alerting/unified/components/notification-policies/EditNotificationPolicyForm.tsx:194]:
```suggestion
          description="Combine multiple alerts into a single notification by grouping them by the same label values. If empty, it is inherited from the parent policy."
```

File: public/app/features/alerting/unified/components/notification-policies/routeTimingsFields.ts [hunk 4]
```diff
@@ -2,18 +2,18 @@ export const routeTimingsFields = {
   groupWait: {
     label: 'Group wait',
     description:
-      'The waiting time until the initial notification is sent for a new group created by an incoming alert. If empty it will be inherited from the parent policy.',
+      'The waiting time before sending the first notification for a new group of alerts. If empty it will be inherited from the parent policy.',
```
Comment (brendamuir) [inline public/app/features/alerting/unified/components/notification-policies/routeTimingsFields.ts:5]:
```suggestion
      'The wait time before sending the first notification for a new group of alerts. If empty, it is inherited from the parent policy.',
```

File: public/app/features/alerting/unified/components/notification-policies/routeTimingsFields.ts [hunk 5]
```diff
@@ -2,18 +2,18 @@ export const routeTimingsFields = {
...
     ariaLabel: 'Group wait value',
   },
   groupInterval: {
     label: 'Group interval',
     description:
-      'The waiting time to send a batch of new alerts for that group after the first notification was sent. If empty it will be inherited from the parent policy.',
+      'The waiting time before sending a notification about changes in the alert group after the first notification has been sent. If empty it will be inherited from the parent policy.',
```
Comment (brendamuir) [inline public/app/features/alerting/unified/components/notification-policies/routeTimingsFields.ts:11]:
```suggestion
      'The wait time before sending a notification about changes in the alert group after the first notification has been sent. If empty, it is inherited from the parent policy.',
```

File: public/app/features/alerting/unified/components/notification-policies/routeTimingsFields.ts [hunk 6]
```diff
@@ -2,18 +2,18 @@ export const routeTimingsFields = {
...
     ariaLabel: 'Group interval value',
   },
   repeatInterval: {
     label: 'Repeat interval',
-    description: 'The waiting time to resend an alert after they have successfully been sent.',
+    description: 'The waiting time before resending a notification that has already been sent successfully.',
```
Comment (brendamuir) [inline public/app/features/alerting/unified/components/notification-policies/routeTimingsFields.ts:16]:
```suggestion
    description: 'The wait time before resending a notification that has already been sent successfully.',
```

File: public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx [hunk 7]
```diff
@@ -220,20 +209,18 @@ function NeedHelpInfoForContactpoint() {
           <br />
           Notifications for firing alert instances are grouped based on folder and alert rule name.
           <br />
-          The waiting time until the initial notification is sent for a new group created by an incoming alert is 30
-          seconds.
+          The waiting time before sending the first notification for a new group of alerts is 30 seconds.
```
Comment (brendamuir) [inline public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx:212]:
```suggestion
          The wait time before sending the first notification for a new group of alerts is 30 seconds.
```

File: public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx [hunk 8]
```diff
@@ -220,20 +209,18 @@ function NeedHelpInfoForContactpoint() {
...
           <br />
-          The waiting time to send a batch of new alerts for that group after the first notification was sent is 5
-          minutes.
+          The waiting time before sending a notification about changes in the alert group after the first notification
+          has been sent is 5 minutes.
           <br />
-          The waiting time to resend an alert after they have successfully been sent is 4 hours.
+          The waiting time before resending a notification that has already been sent successfully is 4 hours.
```
Comment (brendamuir) [inline public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx:217]:
```suggestion
          The wait time before resending a notification that has already been sent successfully is 4 hours.
```

File: public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/route-settings/RouteSettings.tsx [hunk 9]
```diff
@@ -67,7 +67,7 @@ export const RoutingSettings = ({ alertManager }: RoutingSettingsProps) => {
       {overrideGrouping && (
         <Field
           label="Group by"
-          description="Group alerts when you receive a notification based on labels. If empty it will be inherited from the default notification policy."
+          description="Combine multiple alerts into a single notification by grouping them by the same label values. If empty it will be inherited from the default notification policy."
```
Comment (brendamuir) [inline public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/route-settings/RouteSettings.tsx:70]:
```suggestion
          description="Combine multiple alerts into a single notification by grouping them by the same label values. If empty, it is inherited from the default notification policy."
```

**Review by brendamuir — APPROVED** [review]

**Review by gillesdemey — APPROVED** [review]

**Pull request merged**